col/src/main/java/vct/col/ast/util/ASTFactory.java

```diff
@@ -432,6 +432,27 @@ public ASTNode fold(StandardOperator op, ArrayList<ASTNode> list) {
      }
      return res;
    }
+
+   public IfStatement fold(List<IfStatement> list) {
+     if (list.size() == 0) {
```

Comment:
Do you mean that the else branch (on line 450 below) should be wrapped in a block containing `nextIf`, instead of adding it as a standalone statement? Because I put each nested if in the else block here, so there are no `else if`s here, only `else { if (..) { .. } }` I think.